File: app/src/main/java/com/nan/noisedetector/util/PreferenceHelper.java
package com.nan.noisedetector.util;

/**
 * Created by nan on 2019-11-21.
 */
public class PreferenceHelper {
    private static final String KEY_OPEN_NOTIFY = "key_open_notify";

    public static void setOpenNotify(boolean openNotify) {
        PreferenceUtil.getInstance().putBoolean(KEY_OPEN_NOTIFY, openNotify);
    }

    public static boolean isOpenNotify() {
        return PreferenceUtil.getInstance().getBoolean(KEY_OPEN_NOTIFY, false);
    }

    private static final String KEY_THRESHOLD = "key_threshold";


    public static void setThreshold(float threshold) {
        PreferenceUtil.getInstance().putFloat(KEY_THRESHOLD, threshold);
    }

    public static float getThreshold() {
        return PreferenceUtil.getInstance().getFloat(KEY_THRESHOLD, 20);
    }
}

File: app/src/main/java/com/nan/noisedetector/util/PreferenceHelper.kt
package com.nan.noisedetector.util

import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.nan.noisedetector.bean.DataBean

/**
 * Created by nan on 2019-11-21.
 */
object PreferenceHelper {
    private const val KEY_OPEN_NOTIFY = "key_open_notify"
    var isOpenNotify: Boolean
        get() = PreferenceUtil.getInstance().getBoolean(KEY_OPEN_NOTIFY, false)
        set(openNotify) {
            PreferenceUtil.getInstance().putBoolean(KEY_OPEN_NOTIFY, openNotify)
        }

    private const val KEY_THRESHOLD = "key_threshold"
    var threshold: Float
        get() = PreferenceUtil.getInstance().getFloat(KEY_THRESHOLD, 20f)
        set(threshold) {
            PreferenceUtil.getInstance().putFloat(KEY_THRESHOLD, threshold)
        }

    private const val KEY_HISTORY_RECORD = "key_history_record"
    var historyRecord: ArrayList<DataBean>
        get() =
            with(PreferenceUtil.getInstance().getString(KEY_HISTORY_RECORD, "")) {
                if (this.isEmpty()) {
                    return ArrayList()
                }
                val type = object: TypeToken<List<DataBean>>() {}.type
                return Gson().fromJson(this, type)
            }
        set(value) {
            val json: String = Gson().toJson(value)
            PreferenceUtil.getInstance().putString(KEY_HISTORY_RECORD, json)
        }
}
